client-core/src/wallet/syncer.rs

```diff
@@ -1337,50 +1342,135 @@ mod tests {
+        let ip_port = split_nodeid[1].clone();
+        ip_port.split(':');
+        let split_ip: Vec<String> = ip_port.split(':').map(|a| a.to_string()).collect();
+        if 2 != split_ip.len() {
+            return Err(Error::new(
+                ErrorKind::InvalidInput,
+                format!("invalid nodelist_info ip_port {}", ip_port),
+            ));
+        }
+        let ip = split_ip[0].clone();
+        let port = split_ip[1]
+            .parse::<u32>()
+            .chain(|| (ErrorKind::InvalidInput, "Unable to parse node string"))?;
+        all_nodes.push((nodeid, ip, port));
+    }
+    Ok(all_nodes)
+}
+
+/// FIXME change to str::strip_prefix after toolchain upgraded.
+fn strip_prefix<'a>(s: &'a str, prefix: &str) -> Option<&'a str> {
+    if s.len() >= prefix.len() && &s[0..prefix.len()] == prefix {
+        Some(&s[prefix.len()..])
+    } else {
+        None
+    }
+}
+
+/// FIXME change to str::strip_suffix after toolchain upgraded.
+fn strip_suffix<'a>(s: &'a str, suffix: &str) -> Option<&'a str> {
+    if s.len() >= suffix.len() && &s[s.len() - suffix.len()..s.len()] == suffix {
+        Some(&s[0..s.len() - suffix.len()])
+    } else {
+        None
+    }
+}
+
 /// [new light client design](https://github.com/informalsystems/tendermint-rs/blob/master/docs/architecture/adr-006-light-client-refactor.md)
 pub fn spawn_light_client_supervisor(
     db_path: &Path,
-    addr: &str,
+    // ws://localhost:26657/websocket
+    websocket_addr: &str,
     trusting_period: Duration,
+    light_client_peers_user: String,
 ) -> Result<LightClientWrapper<impl Handle + 'static>> {
-    // convert "ws://host:port/websocket" to "tcp://host:port"
-    let addr = format!(
-        "tcp://{}",
-        strip_prefix(addr, "ws://")
+    let mut light_client_peers = light_client_peers_user.clone();
+    let default_primary = format!(
+        "0000000000000000000000000000000000000000@{}",
+        strip_prefix(websocket_addr, "ws://")
+            .and_then(|addr| strip_suffix(addr, "/websocket"))
+            .err_kind(ErrorKind::InvalidInput, || "invalid tendermint rpc address")?
+    );
+    let default_witness = format!(
+        "1000000000000000000000000000000000000000@{}",
+        strip_prefix(websocket_addr, "ws://")
             .and_then(|addr| strip_suffix(addr, "/websocket"))
             .err_kind(ErrorKind::InvalidInput, || "invalid tendermint rpc address")?
     );
-    let addr = tendermint::net::Address::from_str(&addr).unwrap();
-
-    // FIXME use actual tendermint node id as peer id
-    let primary: PeerId = "BADFADAD0BEFEEDC0C0ADEADBEEFC0FFEEFACADE".parse().unwrap();
-    let witness: PeerId = "CEFEEDBADFADAD0C0CEEFACADE0ADEADBEEFC0FF".parse().unwrap();
 
-    let primary_path = db_path.join(primary.to_string());
-    let witness_path = db_path.join(witness.to_string());
+    if "" == light_client_peers {
+        log::info!("use default primary peer {}", default_primary);
```

Comment:
maybe `warn!` -- or should this just reject? I don't know if using a wrong Peer ID could cause any issues later on (e.g. if one connects to the same address, but uses a different/correct Peer ID)?

Could the Peer ID be fetched from TM RPC before this? e.g. in "status", you get "node_info" where it has "id"?

Reply:
https://docs.tendermint.com/master/rpc/#/Info/status

Reply:
how about reject?
i added default for passing integration test.
because peer id  is to tell between peers, the same address will be ignoreded (from tenderment manual)

Reply:
Inside the light client, peerid seems to be only used for storage indexing -- @yihuang  ?
with one fixed default ID, I guess one can corrupt the storage by mashing up blocks from different peers? or for multiple witnesses, if one is faulty, it'll consider all of them faulty?

maybe reject is a correct behaviour?

Reply:
i'll add rejection if
0000000@127.0.0.1,
0000000@127.1.2.3,
IDs overlap